ETWAnalyzer_uTest/EventDump/DumpFileTests.cs

```diff
@@ -504,6 +519,145 @@ public void MinMaxSizeFilter()
+                {
+                    Process = myCmdProcess2,
+                    Processes = new HashSet<ETWProcess> {myCmdProcess2},
+                    SourceFileName = File3,
+                    FileName = FileName1
+                },
+            };
+
+            return data;
+        }
+
+        static char[] myNewLineSplitChars = Environment.NewLine.ToArray();
+
+        [Fact]
+        public void PrintIsSummaryTotalsNone()
+        {
+            using var testOutput = new ExceptionalPrinter(myWriter);
+            var data = CreateTestData();
+            DumpFile fileDumper = new()
+            {
+                TopN = new SkipTakeRange(),
+                TopNProcesses = new SkipTakeRange(),
+                ShowTotal = TotalModes.None
+            };
+            StringWriter writer = new();
+            Console.SetOut(writer);
+            fileDumper.PrintData(data);
+            testOutput.Add(writer.ToString());
+            string[] lines = writer.ToString().Split(myNewLineSplitChars, StringSplitOptions.RemoveEmptyEntries);
+            Assert.Equal(7, lines.Length);
+        }
+        [Fact]
+        public void PrintIsSummaryTotals()
+        {
+            using var testOutput = new ExceptionalPrinter(myWriter);
+            var data = CreateTestData();
+            DumpFile fileDumper = new()
+            {
+                TopN = new SkipTakeRange(),
+                TopNProcesses = new SkipTakeRange(),
+                ShowTotal = TotalModes.File
+            };
+            StringWriter writer = new();
+            Console.SetOut(writer);
+            fileDumper.PrintData(data);
+            testOutput.Add(writer.ToString());
+            string[] lines = writer.ToString().Split(myNewLineSplitChars, StringSplitOptions.RemoveEmptyEntries);
+            Assert.Equal(8, lines.Length);
+            Assert.Contains("Process Count: 2", lines[7]);
+            Assert.Contains("TotalTime: 0.00000 s", lines[7]);
+            Assert.Contains("File/s Total with 0 accessed file/s", lines[7]);
+        }
+
+        [Fact]
+        public void PrintIsSummaryTotalsNullFileMode()
+        {
+            using var testOutput = new ExceptionalPrinter(myWriter);
+            var data = CreateTestData();
+            DumpFile fileDumper = new()
+            {
+                TopN = new SkipTakeRange(),
+                TopNProcesses = new SkipTakeRange(),
+            };
+            StringWriter writer = new();
+            Console.SetOut(writer);
+            fileDumper.PrintData(data);
+            testOutput.Add(writer.ToString());
+            string[] lines = writer.ToString().Split(myNewLineSplitChars, StringSplitOptions.RemoveEmptyEntries);
+            Assert.Equal(8, lines.Length);
+            Assert.Contains("Process Count: 2", lines[7]);
+            Assert.Contains("TotalTime: 0.00000 s", lines[7]);
+            Assert.Contains("File/s Total with 0 accessed file/s", lines[7]);
+        }
+
```

Comment:
We should have also a test where ShowTotal = null and only one row is printed where totals are then also omitted. Like a test case OmitSummaryIfOnlyOneLineIsPrinted()